source/fab/steps/analyse.py

```diff
@@ -167,6 +177,21 @@ def run(self, artefact_store: Dict, config):
         analysed_files = self._parse_files(files=files)
         self._add_manual_results(analysed_files)
 
+        # shall we search the results for fortran programs and a c function called main?
+        if self.find_programs:
+            # find fortran programs
+            sets_of_programs = [af.program_defs for af in by_type(analysed_files, AnalysedFortran)]
+            self.root_symbols = list(chain(*sets_of_programs))
+
+            # find c main()
+            c_with_main = list(filter(lambda c: 'main' in c.symbol_defs, by_type(analysed_files, AnalysedC)))
```

Comment:
It may not be enough just to check for symbols called "main". I think we should be okay as class methods called "main" will get mangled but can you have a global variable called "main"? We don't necessarily need to handle these issues in this change but we should know what the possible failure modes are.

Reply:
Agreed, although that feels like an extreme edge case.